File: src/main/kotlin/Java/Escuela/Estudiante.java
package Java.Escuela;

import java.util.ArrayList;
import java.util.List;

class Estudiante {
    private String nombre;
    private List<Curso> cursos;

    public Estudiante(String nombre) {
        this.nombre = nombre;
        this.cursos = new ArrayList<>();
    }

    public String getNombre() {
        return nombre;
    }

    public void agregarCurso(Curso curso) {
        cursos.add(curso);
        curso.agregarEstudiante(this);
    }

    public double calcularPromedio() {
        double suma = 0;
        int cantidad = 0;
        for (Curso curso : cursos) {
            suma += curso.getCalificacion(this);
            cantidad++;
        }
        return cantidad > 0 ? suma / cantidad : 0;
    }

    @Override
    public String toString() {
        return nombre;
    }
}

File: src/main/kotlin/kotlinOPP/Escuela/Estudiante.kt
package kotlinOPP.Escuela

class Estudiante(val nombre: String) {
    private val cursos: MutableList<Curso> = mutableListOf()

    fun agregarCurso(curso: Curso) {
        cursos.add(curso)
        curso.agregarEstudiante(this)
    }

    fun calcularPromedio(): Double {
        var suma = 0.0
        var cantidad = 0
        for (curso in cursos) {
            suma += curso.getCalificacion(this)
            cantidad++
        }
        return if (cantidad > 0) suma / cantidad else 0.0
    }

    override fun toString(): String {
        return nombre
    }
}